draco/fact_utils.py

```diff
@@ -82,7 +92,47 @@ def dict_to_facts(
                 yield make_fact(FactKind.ATTRIBUTE, (path, parent, data))
 
 
-def facts_to_dict(facts: List) -> Mapping:
-    """A generic decoder that converts an answer set into a nested data structure"""
-    pass  # TODO: https://github.com/cmudig/draco2/issues/24
-    return {}
+def get_value(symbol: Symbol):
+    """Get the value of a Clingo symbol."""
+    if symbol.type == SymbolType.Number:
+        return symbol.number
+    elif symbol.type == SymbolType.String or symbol.type == SymbolType.Function:
+        return symbol.name
+
+
+def collect_children(name: str, collector: dict):
```

Comment:
Should be the same as https://github.com/cmudig/draco2/pull/29/files#diff-45dfe05eeac9fb96cecf5b758a46e5492d812272497bfbd13accd30080528e7cR83